feature/security/gnsi/credentialz/tests/ssh_public_key_authentication/ssh_public_key_authentication_test.go

```diff
@@ -0,0 +1,362 @@
+package ssh_public_key_authentication
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/gnsi/credentialz"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	username                    = "testuser"
+	password                    = "i$V5^6IhD*tZ#eg1G@v3xdVZrQwj"
+	userPrivateKeyFilename      = "user"
+	userPublicKeyFilename       = "user.pub"
+	authorizedKeysListVersion   = "v1.0"
+	authorizedKeysListCreatedOn = 1705962293
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func prepareUserKey(t *testing.T, dir string) {
+	caCmd := exec.Command(
```

Comment:
why would this be a "caCmd" as it is not related to a certificate authority.

Reply:
Fixed